google/oauth2/credentials.py

```diff
@@ -195,14 +223,50 @@ def from_authorized_user_file(cls, filename, scopes=None):
             filename (str): The path to the authorized user json file.
             scopes (Sequence[str]): Optional list of scopes to include in the
                 credentials.
-
         Returns:
-            google.oauth2.credentials.Credentials: The constructed
-                credentials.
-
+            google.oauth2.credentials.Credentials: The constructed credentials.
         Raises:
             ValueError: If the file is not in the expected format.
         """
         with io.open(filename, 'r', encoding='utf-8') as json_file:
             data = json.load(json_file)
             return cls.from_authorized_user_info(data, scopes)
+
+    def to_json(self, strip=None, indent=None):
+        # Heavily modified from https://github.com/googleapis/oauth2client/blob/50d20532a748f18e53f7d24ccbe6647132c979a9/oauth2client/client.py#L241 # noqa: E501
+        """Utility function that creates JSON repr. of a Credentials object.
+
+        Args:
+            strip (Sequence[str]): Optional list of members to exclude from the
+                                   generated JSON.
+            indent (int): This parameter is passed to `json.dumps` to create
+                          indentation in the output JSON data.
+
+        Returns:
+            str: A JSON representation of this instance, suitable to pass to
+                 from_json().
+        """
+
+        # The previous code (oauth2client's client.py's `_to_json`) was very
+        # complicated for I believe no benefit.
+        # I have also removed the `to_serialize` capability from the old method
+        # as I do not see the point in it.
+
+        prep = {
+                'token':            self.token,
+                'refresh_token':    self.refresh_token,
+                'token_uri':        self.token_uri,
+                'client_id':        self.client_id,
+                'client_secret':    self.client_secret,
+                'scopes':           self.scopes,
+                'expiry':           self.expiry.timestamp() if self.expiry
```

Comment:
Is this the correct timestamp format?

Reply:
As per previous comment, dropping this for now (or maybe forever!).